qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -166,6 +170,7 @@ def _default_options(cls) -> Options:
         options.x_key = "xval"
         options.result_parameters = []
         options.extra = {}
+        options.fit_method = "least_squares"
```

Comment:
The default should probably be left up to the lmfit library, which will have its own default. Unless you don't want to use their default

Reply:
This is by design of LMFIT. It defines hierarchy in options. Basically, the minimizer is called with "what minimizer do you want to use" and "what options do you want to pass to the minimizer". This is indeed the same with `fitter` and `fitter_options` in tomography analysis.

Reply:
Good point. The default uses LM algorithm and `least_squares` uses Trust Region Reflective algorithm. As far as I observed in the unittests, LM algorithm fails in some unittests with current initial guesses. Of course the selection of fitter and guess are user matter, but for backward compatibility point of view, we should use `least_squares` because this is what the scipy curve fit internally calls. I don't know why LMFIT defaults to `leastsq`.